### PR TITLE
NVO no longer can be used with binoculars/optics

### PR DESCRIPTION
**Pull request by ihatethisengine**
# About the pull request
Original NVG wouldn't allow you to use them with binos, but when Morrow introduced helmet optics, for some reason he didn't carry over the limitation. I don't know if it was intended or simply was an oversight. Sadly my revert to NVGs (https://github.com/cmss13-devs/cmss13/pull/4807) didn't go through, but I still believe that I was right about points I brought up in the PR, so I am going to address them bit by bit now.

# Explain why it's good for the game
Using nightvision with anything that increases your field of view gives an unreasonable advantage which should be limited only to specs and SGs.

# Testing Photographs and Procedure

<details>
I did test it
</details>

# Changelog

:cl: ihatethisengine
balance: night vision optics no longer can be used with binoculars.
/:cl:

### CODE REVIEW

**Comment by irakligabunia** [comment]
>Original NVG wouldn't allow you to use them with binos, but when Morrow introduced helmet optics, for some reason he didn't carry over the limitation

I'm pretty sure its because NVO's have way worse duration

**Comment by ihatethisengine** [comment]
NVO has the same duration, but a lot more inconvenient to recharge (I may address this later)

**Comment by morrowwolf** [comment]
This was not an oversight.

> NVO has the same duration, but a lot more inconvenient to recharge (I may address this later)

Incorrect.

Do with that what you will.

**Comment by ihatethisengine** [comment]
What exactly is incorrect? 
NVG have 2500 max charge and spend 8 per second, 2500÷8=312.5÷60=5.20 minutes
NVO have 10000 max charge and spend 33 per second, 10000÷33=303.03÷60=5.05 minutes
The difference is literally 9.5 seconds, not that much.

**Comment by morrowwolf** [comment]
Let me be more precise:
It is *dishonest* to categorize them as even remotely similar in regard to charge duration or usage time. You could take 1-2 giant cells and have your NVGs on pretty much 24/7 for a round. Walking past this by just saying it is "a lot more inconvenient to recharge" is blatantly sidestepping extremely pertinent information on past usage which was left out out of the new item on purpose.

Regarding design thoughts:
Overall I had seriously considered axing the optics as well as the goggles for standard use. SGs would also lose their night vision but would get something on their HUD for tracking targets in the dark. Having it just be an in-built MD feels lazy when you could make it so much cooler with client images and such. Darkness should be scary for marines for both gameplay and thematic reasons.

Secondary to that, giving engineers the ability to create on the fly rechargers via APCs was another thought for making power in general more fun to add to items and play with while giving engineers more utility. While possibly annoying at times consider possible recharge requirements for sentries or flashlights. Inventory and gear management is a fairly unique and important part of the overall game and I think it'd benefit from it.

*body armor light flickers*
"Your light is running low on charge..."
Oh shit...

Hunt that dread feeling.

Xeno main out 🧑‍🚀

**Comment by Diegoflores31** [comment]
> Let me be more precise: It is _dishonest_ to categorize them as even remotely similar in regard to charge duration or usage time. You could take 1-2 giant cells and have your NVGs on pretty much 24/7 for a round. Walking past this by just saying it is "a lot more inconvenient to recharge" is blatantly sidestepping extremely pertinent information on past usage which was left out out of the new item on purpose.

but old NVGs , could not use binoculats , had very clear visuals , could be broken in 3 slashes , you had to use your storage to carry cells and lost durability each time the got broken.

File: code/game/objects/items/devices/helmet_visors.dm [hunk 1]
```diff
@@ -271,6 +273,10 @@
 	if(!.)
 		return
 
+	if(user.client.view > 7)
+		to_chat(user, SPAN_WARNING("You cannot use \the [src] while using optics."))
```
Comment (Drulikar) [inline code/game/objects/items/devices/helmet_visors.dm:277]:
```suggestion
		to_chat(user, SPAN_WARNING("You cannot use [src] while using optics."))
```

**Review by Drulikar — CHANGES_REQUESTED** [review]
Minor, but you should see if you can also have `/obj/item/clothing/head/helmet/marine/proc/cycle_huds(mob/user)` check whether `can_toggle` returns true with something like
```js
if(!next_visor.can_toggle(user))
	iterator++
	skipped_hud = TRUE
	continue
```

and to the second loop: 
```js
if(!new_visor.can_toggle(user))
	continue
```

Otherwise you get into this state where the visor was selected, but you had the message it won't work, and it won't work when you leave the view again.

image

Alternatively you had the visor try to activate when the view changes back to normal.

Also would be nice if the visors spit out a to_chat if there's nothing to swap to, but up to you if you want to improve that (it'd probably just be a to_chat before the first return FALSE and the final return FALSE in that same proc).

**Comment by Drulikar** [comment]
This plus the other code suggestion should address all my concerns.
[patch.diff.txt](https://github.com/cmss13-devs/cmss13/files/14845473/patch.diff.txt)

**Review by Drulikar — APPROVED** [review]

**Pull request merged**